src/eduid/workers/am/tests/test_tasks.py

```diff
@@ -193,14 +194,22 @@ def test_unverify_duplicate_multiple_attribute_values(self) -> None:
 
     def test_create_locked_identity(self) -> None:
         user_id = ObjectId("901234567890123456789012")  # johnsmith@example.org / babba-labba
-        attributes = {"$set": {"nins": [{"verified": True, "number": "200102031234", "primary": True}]}}
+        attributes = {
+            "$set": {
+                "identities": [{"identity_type": IdentityType.NIN.value, "number": "200102031234", "verified": True}]
+            }
+        }
         new_attributes = check_locked_identity(self.amdb, user_id, attributes, "test")
 
+        # check_locked_identity should create a locked_identity so add it to the expected attributes
         locked_nin = NinIdentity(number="200102031234", created_by="test", is_verified=True)
         locked_identities = LockedIdentityList(elements=[locked_nin])
         attributes["$set"]["locked_identity"] = locked_identities.to_list_of_dicts()
```

Comment:
check_locked_attributes used to modify (and return) attributes, i.e. the same entity, so even if it did not add locked_identity the test was always true as attributes and new_attributes were the same dict